### DISCUSSION TITLE
To realize the function of dynamic dragging line

### DISCUSSION THREAD

**Discussion by Cat202010804010** [bcakmakoglu/vue-flow#1679] [Ideas]
Can you implement the function of dynamic drag line? Similar to the picture below
![20241107-162901](https://github.com/user-attachments/assets/09a1a2c3-ae7c-4484-be7f-8cacd18c37c6)

Clicking anywhere on the line generates a drag node, dragging this node can make the path of the line change dynamically。

**Comment by bcakmakoglu** [comment]
Yeah you can implement that but there's no example for it at the moment.
As you suggested yourself, you'll just have to add an intermediate node for it that allows for the path to be modified.

**Reply by Cat202010804010** [reply]
My idea is to get the path of the clicked line, and dynamically set the necessary points above, and the position of the necessary points is the coordinates of the mouse movement from the beginning to the end. Is there any way to get and set the path of the clicked line?

**Reply by Cat202010804010** [reply]
Now I can use GetPointerPosition method to get the xy position of the mouse dragged, but also to get the instance of the dragged line, is there any way to set the path of the line and redraw this line?

**Comment by Cat202010804010** [comment]
When I use the BaseEdge component, path is passed dynamically,:path="path", where path is computed, depending on sourceX, sourceY, etc., but when I move nodes, path does not change according to sourceX and sourceY changes, how do I manually pass path and keep it responsive?

**Reply by bcakmakoglu** [reply]
Use a computed property like `const edgePath = computed(() => getBezierPath(props))` and pass it as `<BaseEdge :path="edgePath[0] ...>`
